### PR TITLE
[move source language] Fix block inlining

### PR DESCRIPTION
**Pull request by tnowacki**
- Fixed labels for block inlining

## Motivation

- Fix block inlining for #4853 

## Test Plan

- new test (tested with #4853)

### CODE REVIEW

File: language/move-lang/src/cfgir/inline_blocks.rs [hunk 1]
```diff
@@ -85,5 +89,37 @@ fn inline_single_target_blocks(
...
+///   s.t. A is the only one jumping to B
+///
+/// After:
+///   B: block_a; block_b
+fn remap_to_last_target(
+    mut remapping: BTreeMap<Label, Label>,
+    start: Label,
+    blocks: &mut BasicBlocks,
+) {
+    remapping.remove(&start);
```
Comment (bob-wilson) [inline language/move-lang/src/cfgir/inline_blocks.rs:114]: It might be worth a comment here to mention why the start block doesn't need to be relabeled.

Comment (tnowacki) [reply language/move-lang/src/cfgir/inline_blocks.rs:114]: It's more that it *can't* be relabeled due to limitations on the current cfg api. I'll add a comment

**Pull request closed without merging**